src/main/java/io/quarkus/search/app/quarkusio/QuarkusIO.java

```diff
@@ -74,66 +77,188 @@ public void close() throws Exception {
         try (var closer = new Closer<Exception>()) {
             closer.push(Git::close, git);
             closer.push(CloseableDirectory::close, directory);
-            closer.push(Map::clear, guidesMetadata);
         }
     }
 
     @SuppressWarnings("resource")
     public Stream<Guide> guides() throws IOException {
-        return guideDirectories()
-                .flatMap(uncheckedIO(guidesDirectory -> Files.list(guidesDirectory.path)
-                        .filter(path -> {
-                            String filename = path.getFileName().toString();
-                            return !filename.startsWith("_") && !FilenameUtils.getBaseName(filename).equals("README")
-                                    && FilenameUtils.isExtension(filename, "adoc");
-                        })
-                        .map(path -> parseGuide(guidesDirectory, path))));
+        return Stream.concat(versionedGuides(), legacyGuides());
     }
 
-    @SuppressWarnings("resource")
-    private Stream<GuidesDirectory> guideDirectories() throws IOException {
-        return Stream.concat(
-                Stream.of(new GuidesDirectory(QuarkusVersions.LATEST, directory.path().resolve("_guides"))),
-                Files.list(directory.path().resolve("_versions"))
-                        .map(p -> {
-                            var version = p.getFileName().toString();
-                            return new GuidesDirectory(version, p.resolve("guides"));
-                        }));
-    }
-
-    record GuidesDirectory(String version, Path path) {
-    }
-
-    private Guide parseGuide(GuidesDirectory guidesDirectory, Path path) {
-        var guide = new Guide();
-        guide.version = guidesDirectory.version;
-        guide.origin = QUARKUS_ORIGIN;
-        String name = FilenameUtils.removeExtension(path.getFileName().toString());
-        guide.url = httpUrl(webUri, guidesDirectory.version, name);
-        guide.htmlFullContentProvider = new GitInputProvider(git, pagesTree, htmlPath(guidesDirectory.version, name));
-        getMetadata(guidesDirectory).accept(path, guide);
-        return guide;
+    // guides based on the info from the _data/versioned/[version].index/
+    // may contain quarkus.yaml as well as quarkiverse.yml
+    private Stream<Guide> versionedGuides() throws IOException {
+        return Files.list(directory.path().resolve("_data").resolve("versioned"))
+                // we should remove this version form quarkus.io as it is not available on the site + it does not have any rendered html files:
+                .filter(v -> !v.getFileName().toString().equals("3-0"))
```

Comment:
👀  🆗 /⛔ ? 😃

Reply:
https://github.com/quarkusio/quarkusio.github.io/pull/1848

Reply:
Now, if not having rendered guides is a problem, maybe it should be tested and a warning being logged?

Because it would be a pity to break indexing because something like that is in the way.